src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+
+export function useLocalStorage<T>(key: string, initialValue: T | (() => T), ttl: number) {
+  
+  const [value, setValue] = useState<T>(() => {
+    const jsonValue = localStorage.getItem(key)
+
+    if (jsonValue != null) {
+      const item = JSON.parse(jsonValue)
+      if (Date.now() > item.expiry) {
+        localStorage.removeItem(key)
+        return null
+      }
+      return item.value
+    }
+    
+    if (typeof initialValue === 'function') {
+      return (initialValue as ()=>T)()
```

Comment:
Could you please add white spaces to () => T?

Reply:
All related to code styling things such as spacing, indentation, and quotes, it has to be done with Prettier which I use in my IDE as an extension.
Need to understand why it happened.
This is a signal to me I need to pay more attention to styling and less rely on the extension tools.